process/block/preprocess/miniBlockBuilder.go

```diff
@@ -147,51 +147,52 @@ func (mbb *miniBlocksBuilder) updateAccountShardsInfo(tx *transaction.Transactio
 
 // function returns through the first parameter if the given transaction can be added to the miniBlock
 // second return values returns an error in case no more transactions can be added to the miniBlocks
-func (mbb *miniBlocksBuilder) checkAddTransaction(wtx *txcache.WrappedTransaction) (canAddTx bool, canAddMore bool, tx *transaction.Transaction) {
+func (mbb *miniBlocksBuilder) checkAddTransaction(wtx *txcache.WrappedTransaction) (canAddTx bool, canAddMore bool, shouldAddToRemaining bool, tx *transaction.Transaction) {
 	tx, ok := wtx.Tx.(*transaction.Transaction)
 	if !ok {
 		log.Debug("wrong type assertion",
 			"hash", wtx.TxHash,
 			"sender shard", wtx.SenderShardID,
 			"receiver shard", wtx.ReceiverShardID)
-		return false, true, nil
+		return false, true, false, nil
 	}
 
 	if !mbb.haveTime() {
 		log.Debug("time is out")
-		return false, false, tx
+		return false, false, true, tx
 	}
 
 	receiverShardID := wtx.ReceiverShardID
 	miniBlock, ok := mbb.miniBlocks[receiverShardID]
 	if !ok {
 		log.Debug("mini block is not created", "shard", receiverShardID)
-		return false, true, tx
+		return false, true, false, tx
```

Comment:
Actually this check is also done in scheduled, so it will not pass also there. Anyway this error could not appear as in the init method all the miniblocks combination are created: 0->0, 0->1, 0->2 and 0->meta